### PR TITLE
Adding Oneformer HF Wrapper

### PR DESCRIPTION
**Pull request by aaronrockmenezes**
## Description

Adding HF Wrapper for OneFormer model

Needed to work on Batching since there was no Processor support in dc
Only Semantic Segmentation works

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/oneformer.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+            self.model_config)
+
+        self.model_processor.image_processor.num_text = self.model.config.num_queries - self.model.config.text_encoder_n_ctx
+
+        super().__init__(model=self.model,
+                         task=self.task,
+                         tokenizer=None,
+                         **kwargs)
+
+    def _prepare_batch(self, batch: Tuple[Any, Any, Any]):
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/oneformer.py:62]: Review

File: deepchem/models/torch_models/oneformer.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+                self._optimizer_for_vars[var_key] = (optimizer, lr_schedule)
+        time1 = time.time()
+
+        # Main training loop.
+
+        for batch in generator:
+            if restore:
+                self.restore()
+                restore = False
+            inputs: OneOrMany[torch.Tensor]
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/oneformer.py:190]: Review

File: deepchem/models/torch_models/oneformer.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+            last_avg_loss = avg_loss
+
+        if checkpoint_interval > 0:
+            self.save_checkpoint(max_checkpoints_to_keep)
+
+        time2 = time.time()
+        logger.info("TIMING: model fitting took %0.3f s" % (time2 - time1))
+        return last_avg_loss
+
+    def _predict(self, generator: Iterable[Tuple[Any, Any, Any]],
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/oneformer.py:258]: Review. Can I ignore the last 3 args?

File: deepchem/models/torch_models/oneformer.py [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+from typing import Union, Dict, Tuple, Iterable, List, Optional, Callable, Any
+
+from deepchem.models.optimizers import LearningRateSchedule
+from deepchem.trans import Transformer, undo_transforms
+from deepchem.utils.typing import LossFn, OneOrMany
+from deepchem.models.torch_models import HuggingFaceModel
+
+from transformers import AutoProcessor, AutoModelForUniversalSegmentation, OneFormerConfig
+
+logger = logging.getLogger(__name__)
```
Comment (shreyasvinaya) [inline deepchem/models/torch_models/oneformer.py:17]: please add the torch and hf imports under a try-except block, this should resolve the test failuers

File: deepchem/models/torch_models/oneformer.py [hunk 5]
```diff
@@ -0,0 +1,459 @@
+from PIL import Image
```
Comment (shreyasvinaya) [inline deepchem/models/torch_models/oneformer.py:1]: please fix linting on all of your files

**Comment by shreyasvinaya** [comment]
the model seems to be generating `The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled.` error on the torch tests, could you please take a look at it

**Pull request closed without merging**